main/components/ButtonGroup/index.tsx

```diff
@@ -1,137 +1,103 @@
-    };
-
-    if (data.length === 1) {
-      return fullWidthAndRadius;
-    }
-
-    if (isFirst) {
-      return borderForFirstElement;
-    }
+  onPress?: (index: number, item: T) => void;
+  touchableOpacityProps?: TouchableOpacityProps;
+  borderStyle?: {
+    color?: string;
+    width?: number;
+    radius?: number;
+  };
+};
 
-    if (isLast) {
-      if (data.length === 2) {
-        return {
-          ...borderForLastElement,
-          borderLeftWidth: undefined,
-        };
-      }
+interface MultiSelect {
+  selectedIndices?: number[];
+}
 
-      return borderForLastElement;
-    }
+interface SingleSelect {
+  selectedIndex?: number;
+}
 
-    return borderForMiddleElement;
-  };
+export type ButtonGroupRenderItem<ItemT> = (info: {
+  item: ItemT;
+  selected: boolean;
+  index: number;
+}) => React.ReactElement;
+
+const checkSelected = (
+  props: SingleSelect | MultiSelect,
+  index: number,
+): boolean => {
+  if ('selectedIndices' in props) {
+    return props.selectedIndices?.includes(index) || false;
+  }
+
+  if ('selectedIndex' in props) {
+    return props.selectedIndex === index;
+  }
+
+  return false;
+};
 
+export function ButtonGroup<T>({
+  testID,
+  styles,
+  data,
+  renderItem,
+  onPress,
+  touchableOpacityProps,
+  vertical,
+  borderStyle: {
+    width: borderWidth = 0,
+    color: borderColor,
+    radius: borderRadius = 10,
+  } = {},
+  ...props
+}: ButtonGroupProps<T>): React.ReactElement {
   return (
     <View
       testID={testID}
-      style={[{borderColor: color}, styles?.container, style]}
+      style={StyleSheet.flatten([
+        {borderRadius, borderColor, borderWidth, overflow: 'hidden'},
+        styles?.container,
+        vertical ? {flexDirection: 'column'} : {flexDirection: 'row'},
+      ])}
     >
-      {data.map((text, i) => {
+      {data.map((item, index) => {
+        const selected = checkSelected(props, index);
+
         return (
           <TouchableOpacity
-            key={i}
-            activeOpacity={0.85}
-            style={{flex: 1}}
-            onPress={(): void => {
-              if (onPress) {
-                onPress(i);
-              }
-            }}
+            disabled={!onPress}
+            {...touchableOpacityProps}
+            onPress={() => onPress && onPress(index, item)}
+            key={index}
           >
             <View
-              testID={`CHILD_${i}`}
+              testID={`button-group-item-${index}`}
               style={StyleSheet.flatten([
-                selectedIndex === i
-                  ? {...styles?.selectedButton, backgroundColor: color}
-                  : {...styles?.button, borderColor: color},
-                borderWidthAndRadius(i),
-                {borderColor: color},
+                styles?.button,
+                index === data.length - 1
+                  ? {}
+                  : {
+                      borderRightWidth: borderWidth,
+                      borderColor,
+                    },
```

Comment:
Unlike components, it is safe for`style` to be undefined, so you can shorten this as below.
```
index === data.length - 1 && {
  borderRightWidth: borderWidth,
  borderColor,
}
```

Reply:
This could be great if we could find a way for `stroke alignment`. I also did not find this one yet so let's keep it this way.

Reply:
I looked up for using `stroke alignment` but I could not find a way to do so. 🥲
I'll try later when I have time to deep dive into svg! 
Thanks for inspiration.